sources/TileDB.CSharp/Enumeration.cs

```diff
@@ -0,0 +1,330 @@
+        /// <param name="dataType">The data type of the enumeration. Defaults to the
+        /// default data type of <typeparamref name="T"/>.</param>
+        /// <exception cref="InvalidOperationException"><paramref name="dataType"/> has a different
+        /// size from <typeparamref name="T"/>.</exception>
+        public static Enumeration Create<T>(Context ctx, string name, bool ordered, ReadOnlySpan<T> values, ReadOnlySpan<ulong> offsets, DataType? dataType = null) where T : struct
+        {
+            DataType dataTypeActual = dataType ??= EnumUtil.TypeToDataType(typeof(T));
+            if (EnumUtil.DataTypeToType(dataTypeActual) != typeof(T))
+            {
+                ThrowHelpers.ThrowTypeMismatch(dataTypeActual);
+            }
+
+            EnumerationHandle handle;
+            fixed (T* valuesPtr = &MemoryMarshal.GetReference(values))
+            fixed (ulong* offsetsPtr = &MemoryMarshal.GetReference(offsets))
+            {
+                handle = EnumerationHandle.Create(ctx, name, (tiledb_datatype_t)dataTypeActual, VariableSized, ordered, (byte*)valuesPtr, (ulong)values.Length * (ulong)sizeof(T), offsetsPtr, (ulong)offsets.Length * sizeof(ulong));
+            }
+            return new Enumeration(ctx, handle);
+        }
+
+        /// <summary>
+        /// Creates an <see cref="Enumeration"/> from a collection of strings.
+        /// </summary>
+        /// <param name="ctx">The <see cref="Context"/> to use.</param>
+        /// <param name="name">The name of the enumeration.</param>
+        /// <param name="ordered">Whether the enumeration should be considered as ordered.
+        /// If false, prevents inequality operators in <see cref="QueryCondition"/>s from
+        /// being used with this enumeration.</param>
+        /// <param name="values">The collection of values for the enumeration.</param>
+        /// <param name="dataType">The data type of the enumeration. Must be a string type.
+        /// Optional, defaults to <see cref="DataType.StringUtf8"/>.</param>
+        public static Enumeration Create(Context ctx, string name, bool ordered, IReadOnlyCollection<string> values, DataType dataType = DataType.StringUtf8)
+        {
+            using MarshaledContiguousStringCollection marshaledStrings = new(values, dataType);
+            var handle = EnumerationHandle.Create(ctx, name, (tiledb_datatype_t)dataType, VariableSized,
+                ordered, marshaledStrings.Data, marshaledStrings.DataCount, marshaledStrings.Offsets, marshaledStrings.OffsetsCount);
+            return new Enumeration(ctx, handle);
+        }
+
+        /// <summary>
+        /// Disposes the <see cref="Enumeration"/>.
+        /// </summary>
+        public void Dispose()
+        {
+            _handle.Dispose();
+        }
+
+        /// <summary>
+        /// Returns the number of values for each member of the <see cref="Enumeration"/>.
```

Comment:
Please clarify this, and also add a note that this is equivalent to cell_val_num

Reply:
Done.